arelle/plugin/validate/HMRC/__init__.py

```diff
@@ -265,6 +272,8 @@ def checkFacts(facts):
                     factLocalName = f.qname.localName
                     if factLocalName in mandatoryItems[val.txmyType]:
                         mandatoryFacts[factLocalName] = f
+                    if val.txmyType == "charities" and factLocalName in mustHaveOneItem["charities"]:
```

Comment:
Could make this more generic with
```suggestion
                    if val.txmyType in mustHaveOneItem and factLocalName in mustHaveOneItem[val.txmyType]:
```